src/tests/ftest/rebuild/container_rf_enforce.yaml

```diff
@@ -0,0 +1,51 @@
+hosts:
+  test_servers:
+    - server-A
+    - server-B
+    - server-C
+    - server-D
+    - server-E
+    - server-F
+  test_clients:
+    - client-G
+timeout: 360
+server_config:
+  name: daos_server
+  servers:
+    targets: 2
+pool:
+  mode: 511
+  name: daos_server
+  scm_size: 1GB
+  svcn: 3
+  control_method: dmg
+  pool_query_timeout: 30
+container:
+  control_method: daos
+  akey_size: 5
+  dkey_size: 5
+  data_size: 5
+  object_qty: 10
+  record_qty: 10
+  cont_property: !mux
+    rf_0:
+      properties: rf:0
+    rf_1:
+      properties: rf:1
+    rf_2:
+      properties: rf:2
+    rf_3:
+      properties: rf:3
+    rf_4:
+      properties: rf:4
+rebuild:
+  traffic_oc: !mux
+    object_class_sx:
+      object_class: OC_SX
+    object_class_rp_2g1:
+      object_class: OC_RP_2G1
+    object_class_rp_3g1:
+      object_class: OC_RP_3G1
+  cont_rf:
```

Comment:
There's quite a bit of code duplication in this test and rebuild/container_redundancy.py.
I think some of the common code might be better if you put them in a common util file or introduce a base class rather than using RbldContRedundancyFactor as a base class.
In fact, I think if you do a mux here for rank (to kill) to cover killing 1, 2, 3, 4 ranks at a time, we can remove rebuild/container_redundancy.py test. Depending on the number of ranks you are killing and the rf used, you would know to expect HEALTHY or UNCLEAN, and this wil give coverage for testing rf 0-4, instead of just rf2 in the other test.

Reply:
This script already import/reusing the methods from class RbldContRedundancyFactor (from rebuild/container_redundancy.py).
Test in rebuild/container_redundancy.py to verify the redundant factor equal or more than 2 when number of server dropped simultaneously to report error/Unclean or Healthy, we can add/mux with more tests for rf:0, rf:1 or rf:3, with various restart server-rank.
This test is to verify the enforcement of redundant factor with different traffic type, rebuild server-rank is constant.
On negative testcase, this test cover the test with the oc-class traffic which is not able to handle by the container, on positive test scenario, it covered all the rf numbers.

Reply:
So lines 90-112 in rebuild/container_rf_enforce.py in this PR is duplicating what rebuild/container_redundancy.py is doing. In fact, most of the code is duplicated.
I'm ok to keep 2 separate tests:
1. one to test rf enforcement with respect to oc-class (this PR) 
2. another test to test with rebuild (container_redundancy.py)

Grouping the tests as above helps with documentation and debugging when things fail. I think the rf=0 rebuild in this PR should be moved to container_redundancy.py.

So let's do this:
1. rename container_rf_enforce.py to cont_rf_oclass_enforcement.py and keep the rf oclass test in this file. This test should be in ftests/container.
2. move the rf=0 rebuild test in this PR to rebuild/container_redundancy.py so we test both rf=0 and rf=2 rebuild in the same test. Rename rebuild/container_redundancy.py to rebuild/rebuild_container_rf.py
3. Put all common code/functions either in a base class (if it makes sense) or in a common util.

Reply:
Please see the new commit which addressed all of the above comments.